thetis/solver2d.py

```diff
@@ -603,8 +621,11 @@ def assign_initial_conditions(self, elev=None, uv=None, tracer=None, sediment=No
             elev_2d.project(elev)
         if uv is not None:
             uv_2d.project(uv)
-        if tracer is not None and self.options.solve_tracer:
-            self.fields.tracer_2d.project(tracer)
+        if self.options.solve_tracer:
+            for label in self.options.tracer_metadata:
+                tracer = tracers.get(label)
+                if tracer is not None:
+                    self.fields[label].project(tracer)
```

Comment:
I guess you could loop the other way:
```python
for label, func in tracers.items():
    assert label in self.options.tracer_metadata, f'Unknown tracer label {label}'  # it's good to warn users
    self.fields[label].project(func)
```
Now if we wish to keep things consistent, i.e. the args are  `uv`, `elev`, `tracer` etc you could do
```
for l, func in tracers.items():
    label = l + '_2d'
    ...
```